avm/ptn/app/container-job-toolkit/modules/deploy_services.bicep

```diff
@@ -0,0 +1,930 @@
+        '/providers/Microsoft.Authorization/roleDefinitions/'
+      )
+      ? roleAssignment.roleDefinitionIdOrName
+      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName))
+  })
+]
+var formattedRegistryRoleAssignments = [
+  for (roleAssignment, index) in (registryRoleAssignments ?? []): union(roleAssignment, {
+    roleDefinitionId: vaultBuiltInRoleNames[?roleAssignment.roleDefinitionIdOrName] ?? (contains(
+        roleAssignment.roleDefinitionIdOrName,
+        '/providers/Microsoft.Authorization/roleDefinitions/'
+      )
+      ? roleAssignment.roleDefinitionIdOrName
+      : subscriptionResourceId('Microsoft.Authorization/roleDefinitions', roleAssignment.roleDefinitionIdOrName))
+  })
+]
+// network related variables
+var privateEndpointSubnetAddressPrefix = cidrSubnet(addressPrefix, 24, 0) // the first /24 subnet in the address space
+var deploymentscriptSubnetAddressPrefix = cidrSubnet(addressPrefix, 24, 1) // the second /24 subnet in the address space
+var workloadSubnetAddressPrefix = cidrSubnet(addressPrefix, 23, 1) // the second /23 subnet in the address space, as the first /24 subnet is used for private endpoints
+var zoneRedundant = deployInVnet || (!empty(addressPrefix) && empty(workloadProfiles)) // zoneRedundant is only needed if the environment is deployed in a vnet or for consumption plan if an addressPrefix has been provided
+
+// filter and prepare secrets that need to be added to Key Vault in order to reference them later
+var secrets = [
+  for secret in keyVaultSecrets ?? []: (secret.keyVaultUrl != null)
+    ? {
+        name: last(split(secret.keyVaultUrl ?? 'dummy-for-deployment-validation', '/'))
+        identity: secret.identity
+        value: 'dummy' // dummy needed to pass the validation
+      }
+    : null
+]
+
+// https://learn.microsoft.com/en-us/azure/virtual-network/service-tags-overview#discover-service-tags-by-using-downloadable-json-files
+var regionSpecificServiceTags = {
+  francecentral: 'centralfrance'
+  francesouth: 'southfrance'
+  germanywestcentral: 'germanywc'
+  germanynorth: 'germanyn'
+  norwayeast: 'norwaye'
+  norwaywest: 'norwayw'
+  switzerlandnorth: 'switzerlandn'
+  switzerlandwest: 'switzerlandw'
+  eastusstg: 'usstagee'
+  southcentralusstg: 'usstagec'
+  brazilsoutheast: 'brazilse'
+}
+var locationLowered = toLower(location)
+var regionServiceTag = regionSpecificServiceTags[?locationLowered] ?? locationLowered
+
+// Networking resources
+// -----------------
+module nsg 'br/public:avm/res/network/network-security-group:0.5.0' = if (zoneRedundant) {
+  name: '${uniqueString(deployment().name, location, resourceGroupName)}-nsg'
+  params: {
+    name: 'nsg-${name}'
+    location: location
+    enableTelemetry: enableTelemetry
+    tags: tags
+    lock: lock
+  }
+}
+module nsg_workload_plan 'br/public:avm/res/network/network-security-group:0.5.0' = if (zoneRedundant && !empty(workloadProfiles)) {
```

Comment:
Please leave one empty line before each module declaration for better readability (applies to whole file)